Android Source/mobile/src/main/java/net/steppschuh/sensordatalogger/PhoneActivity.java

```diff
@@ -117,6 +129,54 @@ public void onClick(View v) {
             }
         });
 
+        floatingActionRecordButton = (FloatingActionButton) findViewById(R.id.recordActionButton);
+        floatingActionRecordButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                // request runtime permission
+                if (ContextCompat.checkSelfPermission(getApplicationContext(),
+                        Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                        != PackageManager.PERMISSION_GRANTED) {
+                    if (ActivityCompat.shouldShowRequestPermissionRationale(app.getContextActivity(),
+                            Manifest.permission.WRITE_EXTERNAL_STORAGE)) {
+                    } else {
+                        ActivityCompat.requestPermissions(app.getContextActivity(),
+                                new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE},23
+                        );
+                        return;
+                    }
+                }
+
+                // if user did not grant permission, notify by Toast
+                if (ContextCompat.checkSelfPermission(getApplicationContext(),
```

Comment:
Not sure why this is checked again here. Note that requesting permissions is an asynchronous process, all permission request results will be delivered via a callback method that you have to override (`onRequestPermissionsResult`). See [requesting runtime permissions](https://developer.android.com/training/permissions/requesting).

Reply:
(2/3) Removed the permission checking, good to know that it was not required. :+1: